src/main/scala/io/iohk/ethereum/consensus/ethash/EthashMiner.scala

```diff
@@ -56,12 +56,12 @@ class EthashMiner(
 
   def processMining(): Unit = {
     val parentBlock = blockchain.getBestBlock()
-    val epoch = EthashUtils.epoch(parentBlock.header.number.toLong + 1)
-
+    val blockNumber = parentBlock.header.number.toLong + 1
+    val epoch = EthashUtils.epoch(blockNumber, blockCreator.blockchainConfig.ecip1099BlockNumber.toLong)
```

Comment:
why not accepting `BigInt` there? I mean - if domain uses `BigInt` for block numbers, why shouldn't miner?

Reply:
Because we are using only `Long`s in `EthashUtils`. I am not sure if we do that because of performance or sth else, but I don't want to change that in this pr.